src/RSocketRequester.cpp

```diff
@@ -35,6 +35,7 @@ RSocketRequester::RSocketRequester(
 
 RSocketRequester::~RSocketRequester() {
   LOG(INFO) << "RSocketRequester => destroy";
+  stateMachine_->close(folly::exception_wrapper(), StreamCompletionSignal::CONNECTION_END);
```

Comment:
> Do you observe if this check is failing if you do not call the close() method in here?

The check doesn't fail because we leak the `RSocketStateMachine`, hence we never call `~RSocketStateMachine`.
Do you observe if this check is > Do you observe if this check is failing if you do not call the close() method in here?

The check doesn't fail because we leak the `RSocketStateMachine`, hence we never call `~RSocketStateMachine`.  This is because of the smart pointer cycle of
`FrameTransport <-> RSocketStateMachine`.